File: two_sum.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution {
public:
    
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();

        int high = n-1;
        int low = 0;
        int mid;
        while(high>low){
            mid = (high+low)/2;
            if(nums[mid]> target){
                if(nums[mid+1]<target){
                    return mid;
                }
                high = mid-1; 
            }else if (nums[mid]<target)
            {
                if(nums[mid+1]>target){
                    return mid;
                }
                low = mid+1;
            }
            else{
                return mid;
            }
        }
        
    }
};

File: two_sum.py
from typing import List


class Solution:
    
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {num:i for i,num in enumerate(nums)}
        for i,num in enumerate(nums):
            look_up = target-num
            search = hash_map.get(look_up,-1)
            if search not in [-1, i]:
                return [i,search]
            
    